internal/stream/notification_stream.go

```diff
@@ -0,0 +1,75 @@
+package stream
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/sirupsen/logrus"
+)
+
+//go:generate mockgen -source=notification_stream.go -package=stream -destination=mock_notification_stream.go
+
+type Notifier interface {
+	Notify(ctx context.Context, notifiable common.Notifiable) error
+	Close()
+}
+
+type Envelope struct {
+	Name     string
+	Payload  interface{}
+	Metadata interface{}
+}
+
+type NotificationStream struct {
+	metadata interface{}
+	writer   StreamWriter
+	log      logrus.FieldLogger
+}
+
+func NewNotificationStream(writer StreamWriter, logger logrus.FieldLogger, metadata interface{}) *NotificationStream {
+	return &NotificationStream{
+		writer:   writer,
+		metadata: metadata,
+		log:      logger,
+	}
+
+}
+
+func (s *NotificationStream) Notify(ctx context.Context, notifiable common.Notifiable) error {
+	if s.writer == nil {
+		return nil
+	}
+	if notifiable == nil || reflect.ValueOf(notifiable).IsNil() {
```

Comment:
I don't think you need the reflect bit here.

Reply:
Apparently in Golang if you pass an empty interface the nil check is not enough, I've added a [test just to check the reflection bit](https://github.com/openshift/assisted-service/pull/5013/files/8b2c3565a7dcbe0b9ee26d0ca050da8433a79a86#diff-65c6c20c0876dfb3d2a2e3277f325d97eef102b1d4fe701ab59db0e76f15f2e1R92).